File: packages/drm-plugin/android/src/main/java/com/twg/videodrm/DRMPlugin.kt
package com.twg.videodrm

import com.twg.videodrm.DRMManager.DRMManager
import com.twg.video.core.player.DRMManagerSpec
import com.twg.video.core.plugins.NativeVideoPlayerSource
import com.twg.video.core.plugins.ReactNativeVideoPlugin

class DRMPlugin(name: String) : ReactNativeVideoPlugin(name) {
  override fun getDRMManager(source: NativeVideoPlayerSource): DRMManagerSpec? {
    return DRMManager(source)
  }
}

File: packages/drm-plugin/ios/DRMPlugin.swift
//
//  DRMPlugin.swift
//  ReactNativeVideoDrm
//
//  Created by Krzysztof Moch on 07/08/2025.
//

import Foundation
import ReactNativeVideo

class DRMPlugin: ReactNativeVideoPlugin {
  override func getDRMManager(source: any NativeVideoPlayerSource) -> (any DRMManagerSpec)? {
    #if targetEnvironment(simulator)
      // DRM is not supported on the simulator.
      print("[ReactNativeVideoDRM] DRM is not supported on the simulator. Returning nil.")
      return nil
    #else
      return DRMManager(source: source)
    #endif
  }
}
